src/api/services/auditTrailService.ts

```diff
@@ -54,6 +54,38 @@ export const insertSharingAuditTrails = async (
   }
 };
 
+export const insertManageApiKeyAuditTrail = async (
+  participant: Participant,
+  userId: number,
+  userEmail: string,
+  action: AuditAction,
+  keyName: String,
+  apiRoles: String[],
+  traceId: string
+) => {
+  try {
+    const manageApiKeyTrail: Omit<AuditTrailDTO, 'id'> = {
+      userId,
+      userEmail,
+      event: AuditTrailEvents.ManageApiKey,
+      eventData: {
+        siteId: participant.siteId!,
+        action,
+        apiRoles,
+        keyName,
+        participantId: participant.id,
+      },
+      succeeded: false,
+    };
+
+    return await AuditTrail.query().insert(manageApiKeyTrail);
+  } catch (error) {
+    const { errorLogger } = getLoggers();
```

Comment:
The contents of this `catch` block are repeated many times in this file. Maybe now is a good time to extract it out? What do you think?

Reply:
I think that would be good to do, though this PR is already pretty large so perhaps another PR can be created to do that later? 
Once note though,`insertApproveAccountAuditTrail` doesn't seem to have it?

Reply:
yeah let's raise a separate ticket for this